test/e2e/network/ingressclass.go

```diff
@@ -128,3 +137,148 @@ func deleteIngressClass(cs clientset.Interface, name string) {
+	/*
+		Release: v1.19
+		Testname: IngressClass API
+		Description:
+		- The networking.k8s.io API group MUST exist in the /apis discovery document.
+		- The networking.k8s.io/v1beta1 API group/version MUST exist in the /apis/networking.k8s.io discovery document.
+		- The IngressClasses resources MUST exist in the /apis/networking.k8s.io/v1beta1 discovery document.
+		- The IngressClass resource must support create, get, list, watch, update, patch, delete, and deletecollection.
+	*/
+	ginkgo.It(" should support creating IngressClass API operations", func() {
+
+		// Setup
+		icClient := f.ClientSet.NetworkingV1beta1().IngressClasses()
+		icVersion := "v1beta1"
+
+		// Discovery
+		ginkgo.By("getting /apis")
+		{
+			discoveryGroups, err := f.ClientSet.Discovery().ServerGroups()
+			framework.ExpectNoError(err)
+			found := false
+			for _, group := range discoveryGroups.Groups {
+				if group.Name == networkingv1beta1.GroupName {
+					for _, version := range group.Versions {
+						if version.Version == icVersion {
+							found = true
+							break
+						}
+					}
+				}
+			}
+			framework.ExpectEqual(found, true, fmt.Sprintf("expected networking API group/version, got %#v", discoveryGroups.Groups))
+		}
+		ginkgo.By("getting /apis/networking.k8s.io")
+		{
+			group := &metav1.APIGroup{}
+			err := f.ClientSet.Discovery().RESTClient().Get().AbsPath("/apis/networking.k8s.io").Do(context.TODO()).Into(group)
+			framework.ExpectNoError(err)
+			found := false
+			for _, version := range group.Versions {
+				if version.Version == icVersion {
+					found = true
+					break
+				}
+			}
+			framework.ExpectEqual(found, true, fmt.Sprintf("expected networking API version, got %#v", group.Versions))
+		}
+
+		ginkgo.By("getting /apis/networking.k8s.io" + icVersion)
+		{
+			resources, err := f.ClientSet.Discovery().ServerResourcesForGroupVersion(networkingv1beta1.SchemeGroupVersion.String())
+			framework.ExpectNoError(err)
+			foundIC := false
+			for _, resource := range resources.APIResources {
+				switch resource.Name {
+				case "ingressclasses":
+					foundIC = true
+				}
+			}
+			framework.ExpectEqual(foundIC, true, fmt.Sprintf("expected ingressclasses, got %#v", resources.APIResources))
+		}
+
+		// IngressClass resource create/read/update/watch verbs
+		ginkgo.By("creating")
+		ingressClass1, err := createIngressClass(cs, "ingressclass1", true)
+		framework.ExpectNoError(err)
+		_, err = createIngressClass(cs, "ingressclass2", true)
+		framework.ExpectNoError(err)
+		_, err = createIngressClass(cs, "ingressclass3", true)
+		framework.ExpectNoError(err)
+
+		ginkgo.By("getting")
+		gottenIC, err := icClient.Get(context.TODO(), ingressClass1.Name, metav1.GetOptions{})
+		framework.ExpectNoError(err)
+		framework.ExpectEqual(gottenIC.UID, ingressClass1.UID)
+		framework.ExpectEqual(gottenIC.UID, ingressClass1.UID)
+
+		ginkgo.By("listing")
+		ics, err := icClient.List(context.TODO(), metav1.ListOptions{LabelSelector: "ingressclass=tester"})
```

Comment:
need to filter specifically to the ingressclass objects created for this test, add the same label to all of them, and set up a shared listOptions that selects that label, and use it consistently for list/deletecollection API calls